mdpow/tests/test_workflows_base.py

```diff
@@ -62,17 +59,23 @@ def test_project_paths_csv_input(self, csv_input_data):
 
         pd.testing.assert_frame_equal(project_paths, csv_df)
 
-    def test_automated_project_analysis(self, project_paths_data, caplog):
+    def test_dihedral_analysis_figdir_requirement(self, project_paths_data, caplog):
+        caplog.clear()
+        caplog.set_level(logging.ERROR, logger='mdpow.workflows.base')
+        
         project_paths = project_paths_data
         # change resname to match topology (every SAMPL7 resname is 'UNK')
         # only necessary for this dataset, not necessary for normal use
         project_paths['resname'] = 'UNK'
 
-        base.automated_project_analysis(project_paths, solvents=('water',),
-                                 ensemble_analysis='DihedralAnalysis')
+        with pytest.raises(AssertionError,
+                           match="figdir MUST be set, even though it is a kwarg. Will be changed with #244"):
+        
+            base.automated_project_analysis(project_paths, solvents=('water',),
+                                            ensemble_analysis='DihedralAnalysis')
 
-        assert 'all analyses completed' in caplog.text, ('automated_dihedral_analysis '
-               'did not iteratively run to completion for the provided project')
+            assert 'all analyses completed' in caplog.text, ('automated_dihedral_analysis '
```

Comment:
Unindent — that's better outside the pytest.raises block.

As a general rule reduce the amount of code in a with block to the essentials as this makes clearer what's relevant for testing the condition.

Reply:
I believe the test fails without the indent, possibly because of what is being tested or how it is written in the module. I am not certain of this, but making the change last night resulted in a passing test.